### PR TITLE
use merge_group

### PR DESCRIPTION
**Pull request by morenol**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -152,7 +149,7 @@ jobs:
   done:
     name: Done
     needs: [check_fmt, check_security, smoke_test, macos_ci]
-    if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
+    if: ${{ github.event_name == 'merge_group' }}
```
Comment (morenol) [inline .github/workflows/ci.yml:152]:
```suggestion
    if: github.event_name == 'merge_group'
```

**Review by sehz — APPROVED** [review]
LGTM

**Comment by morenol** [comment]
@sehz we need to update the settings of the repo to enable merge_group

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -152,7 +149,7 @@ jobs:
   done:
     name: Done
     needs: [check_fmt, check_security, smoke_test, macos_ci]
-    if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
+    if: github.event_name == 'merge_group'
```
Comment (morenol) [inline .github/workflows/ci.yml:152]:
```suggestion
```

**Comment by sehz** [comment]
merge group is enabled for this repo

**Pull request merged**